lite2/client.go

```diff
@@ -757,6 +782,65 @@ func (c *Client) fetchHeaderAndValsAtHeight(height int64) (*types.SignedHeader,
 	return h, vals, nil
 }
 
+// fetchMissingTrustedHeader finds the closest height after the
+// requested height and does backwards verification.
+func (c *Client) fetchMissingTrustedHeader(height int64, now time.Time) (*types.SignedHeader, error) {
+	c.logger.Info("Fetching missing header", "height", height)
+
+	trustedHeader, err := c.trustedStore.SignedHeaderAfter(height)
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't get signed header after %d", height)
+	}
+
+	var untrustedHeader *types.SignedHeader
+	for i := trustedHeader.Height - 1; i >= height; i-- {
+		untrustedHeader, err = c.signedHeaderFromPrimary(i)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to obtain the header #%d", i)
+		}
+
+		if err := untrustedHeader.ValidateBasic(c.chainID); err != nil {
+			return nil, errors.Wrap(err, "untrustedHeader.ValidateBasic failed")
+		}
+
+		if untrustedHeader.Height >= trustedHeader.Height {
+			return nil, errors.Errorf("expected older header height %d to be less than one of newer header %d",
+				untrustedHeader.Height,
+				trustedHeader.Height)
+		}
+
+		if !untrustedHeader.Time.Before(trustedHeader.Time) {
+			return nil, errors.Errorf("expected older header time %v to be before newer header time %v",
+				untrustedHeader.Time,
+				trustedHeader.Time)
+		}
+
+		if HeaderExpired(untrustedHeader, c.trustingPeriod, now) {
+			return nil, ErrOldHeaderExpired{untrustedHeader.Time.Add(c.trustingPeriod), now}
+		}
+
+		if !bytes.Equal(untrustedHeader.Hash(), trustedHeader.LastBlockID.Hash) {
+			return nil, errors.Errorf("older header hash %X does not match trusted header's last block %X",
+				untrustedHeader.Hash(),
+				trustedHeader.LastBlockID.Hash)
+		}
```

Comment:
`trustedHeader` remains the same throughout the entire for loop yet `untrustedHeader` changes iteratively - shouldn't trustedHeader be updated to the lastUntrustedHeader in order for the LastBlockID check to be correctly used?